src/shared/clusterutils/username.go

```diff
@@ -0,0 +1,47 @@
+package clusterutils
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"strings"
+)
+
+const (
+	HashedUsernameSectionMaxLength = 17
+)
+
+func BuildHashedUsername(workloadName, namespace, clusterUID string) string {
+	// We're trying to achieve uniqueness while not completely trimming the clusterUID if username + namespace are long
+	// so we trim each section separately leaving room for 6 char hash suffix and string formatting chars
+	username := fmt.Sprintf("%s-%s-%s", workloadName, namespace, clusterUID)
+	hash := md5.Sum([]byte(username))
+
+	if len(username) > HashedUsernameSectionMaxLength {
+		username = username[:HashedUsernameSectionMaxLength]
+	}
+
+	if len(namespace) > HashedUsernameSectionMaxLength {
+		namespace = namespace[:HashedUsernameSectionMaxLength]
+	}
+
+	if len(clusterUID) > HashedUsernameSectionMaxLength {
+		clusterUID = clusterUID[:HashedUsernameSectionMaxLength]
```

Comment:
well, the linter has a point. There's some mess left in this function. 
Consider adding a UT for it to catch this mess.